rust-toolchain.toml

```diff
@@ -7,4 +7,4 @@
 # that the newly specified version is baked in and doesn't have to be installed
 # (and rust deps rebuilt) during every incremental build.
 [toolchain]
-channel = "1.69.0"
+channel = "1.70.0"
```

Comment:
We'll need to bump the nightly version as well. (There probably ought  to be a comment and pointer here; and maybe we ought to make a rust-toolchain-nightly.toml that the corresponding workflows use somehow?)

```
.github/workflows/sanitizers.yml:          TOOLCHAIN=nightly-2023-03-13
ci/container_scripts/install_extra_deps.sh:  rustup override set nightly-2023-03-13
```

Reply:
Good catch! We do list these in our "CI Maintenance" workflow so that we don't forget them, but it's easy to forget them when we're doing it outside of the github issue that the workflow creates. I'm not sure how we could use a custom rust toolchain file though. Maybe we could just store it in a text file and then do something like `rustup override set "$(cat ../../rust-nightly-toolchain)"` or somethng?

Reply:
Or maybe we could do something with symbolic links?

e.g.
* rename `rust-toolchain.toml` to `rust-toolchain-stable.toml`
* add `rust-toolchain-nightly.toml`
* add a symbolic link `rust-toolchain.toml` -> `rust-toolchain-stable.toml`

Then the workflows that need the nightly could just locally reset the symbolic link. That'd make it easier to switch back and forth locally between the "blessed" stable and nightly as well.

If you'd rather not sort it out in this PR, I'm also fine with just changing the version in the current 3 places for the moment :)

Reply:
Okay that sounds like a good idea. For now I just updated the nightly version in this PR, but I'll make the symlink change in another PR.